validate.js

```diff
@@ -1114,6 +1114,48 @@
       if (!PATTERN.exec(value)) {
         return message;
       }
+    },
+
+    // A type validator used to enforce generic types on data.
+    // wraps extant isThing helpers
+    type: function(value, options) {
+      // a/an selector to produce proper grammer in error messages
+      function aOrAn(nextWord) {
+        var vowels = "aeiou";
+        if (( typeof nextWord !== "string" ) || !nextWord.length) {
+          return "a";
+        }
+        return (vowels.indexOf(nextWord[0]) === -1) ? "a" : "an";
+      }
+      var typeHelperMap = {
+        number:   v.isNumber,
+        integer:  v.isInteger,
+        string:   v.isString,
+        date:     v.isDate,
+        array:    v.isArray,
+        object:   v.isObject,
+        function: v.isFunction,
+        promise:  v.isPromise,
+        hash:     v.isHash,
+        bookean:  v.isBoolean
+      };
+      if (!v.isObject(options)) {  // accept options as string or object
+        options = { type: options };
+      }
+      if (value === undefined || value === null) {  // undefined/null checking is already covered by 'presence'
```

Comment:
Doesn't need to be changed again, but just realized you could probably convert this to:
`if (!v.isDefined(value)) {`